XSLT/rhodes_dspace_xoai_to_mods_farnsworth.xsl

```diff
@@ -0,0 +1,327 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+    xmlns:xs="http://www.w3.org/2001/XMLSchema"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns:lyncode="http://www.lyncode.com/xoai"
+    xmlns:dltn = "https://github.com/digitallibraryoftennessee"
+    xmlns="http://www.loc.gov/mods/v3"
+    exclude-result-prefixes="#all"
+    xpath-default-namespace="http://www.lyncode.com/xoai" version="2.0">
+    
+    <!-- output settings -->
+    <xsl:output encoding="UTF-8" method="xml" omit-xml-declaration="yes" indent="yes"/>
+    <xsl:strip-space elements="*"/>
+    
+    <!-- includes and imports -->
+    
+    <!-- Types -->
+    <xsl:param name="pType">
+        <dltn:type string="moving image">Video</dltn:type>
+        <dltn:type string="text">Text</dltn:type>
+        <dltn:type string="sound recording">Sound</dltn:type>
+        <dltn:type string="still image">Still image</dltn:type>
+        <dltn:type string="still image">Image</dltn:type>
+    </xsl:param>
+    
+    
+    <!-- identity transform -->
+    <xsl:template match="@* | node()">
+        <xsl:copy>
+            <xsl:apply-templates select="@* | node()"/>
+        </xsl:copy>
+    </xsl:template>
+    
+    <!-- normalize all the text! -->
+    <xsl:template match="text()">
+        <xsl:value-of select="normalize-space(.)"/>
+    </xsl:template>
+    
+    <!-- match metadata -->
+    <xsl:template match="lyncode:metadata">
+        <!-- match the document root and return a MODS record -->
+        <mods xmlns="http://www.loc.gov/mods/v3" version="3.5"
+            xmlns:xlink="http://www.w3.org/1999/xlink"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://www.loc.gov/mods/v3 http://www.loc.gov/standards/mods/v3/mods-3-5.xsd">
+            
+            <!-- identifiers -->
+            <xsl:apply-templates select='element[@name="dc"]/element[@name="identifier"]/element[@name="other"]/element[@name="none"]/field[@name="value"]'/>
+            
+            <!-- title-->
+            <xsl:apply-templates select="element[@name = 'dc']/element[@name = 'title']/element/field"/>
+            
+            <!-- alternative title -->
+            <xsl:apply-templates select='element[@name="dc"]/element[@name="title"]/element[@name="alternative"]/element[@name="en_US"]/field'/>
+            
+            <!-- abstract -->
+            <xsl:apply-templates select='element[@name="dc"]/element[@name="description"]/element[@name="abstract"]/element[@name="en_US"]/field[@name="value"]'/>
+            
+            <!-- urls -->
```

Comment:
Multiple location elements are problematic for our mapping to DPLA.  This is because our data is stored as JSON-LD and the data type changes from an object to an array with multiple location elements.  Instead, do something like this:

```
<location>
<!-- Handle -->
     <xsl:template select='element[@name = "dc"]/element[@name = "identifier"]/element[@name="uri"]/element[@name="none"]/field[@name = "value"]' />
<!-- Thumbnail -->
     <xsl:template select='element[@name="bundles"]/element[@name="bundle"][field[@name="name"][text()="THUMBNAIL"]]/element[@name="bitstreams"]/element[@name="bitstream"][1]/field[@name="url"]' />
</location>

```

Reply:
Thanks - this makes complete sense.